R/population-workflow.R

```diff
@@ -26,251 +16,441 @@ PopulationWorkflow <- R6::R6Class(
+                                                 pkParametersToEvaluate = NULL,
+                                                 userDefinedPKFunctions = NULL,
+                                                 pkParameterResultsFilePath = file.path(self$pkParametersFolder, "populationPKParameters.csv")) {
+      self$populationPKParameters <- CalculatePKParametersTask$new(
+        input = input,
+        output = output,
+        settings = settings,
+        active = active,
+        message = message %||% "Calculate PK parameters for population",
+        simulationFilePath = simulationFilePath,
+        simulationResultFilePaths = simulationResultFilePaths,
+        pkParametersToEvaluate = pkParametersToEvaluate,
+        userDefinedPKFunctions = userDefinedPKFunctions,
+        pkParameterResultsFilePath = pkParameterResultsFilePath
+      )
+    },
 
+    # TO DO: Define the tasks settings for plots
+    #' @description
+    #' Define plot demography `task` settings
+    #' @param input file or folder of input
+    #' @param output file or folder of output
+    #' @param settings specific settings for task
+    #' @param active logical indicating if `task` is performed in worklfow.
+    #' Default value is `TRUE`
+    #' @param message title of the `task`.
+    #' Default value indicates `task` name.
+    #' @return A new `Task` object
+    plotDemographySettings = function(input = NULL,
+                                          output = NULL,
+                                          active = TRUE,
+                                          message = NULL) {
+      self$plotDemography <- Task$new(
+        input = input,
+        output = output,
+        active = active,
+        message = "Plot Demography task not available at the moment"
+      )
+    },
 
-
-
-
-
-    setDemographyPlotSettings = function(input = NULL,
-                                             output = NULL,
-                                             active = TRUE,
-                                             message = NULL) {
-      self$demographyPlot <- Task$new(
-        input = input %||% self$populationSimulation$input,
-        output = output %||% list(
-          "demographyResults" = file.path(self$simulationFolder, "demography.RData"),
-          "demographyPlot" = file.path(self$outputFolder, "demographyPlot"),
-          "demographyTable" = file.path(self$outputFolder, "demographyTable.md")
-        ),
+    #' @description
+    #' Define plot goodness of fit `task` settings
+    #' @param input file or folder of input
+    #' @param output file or folder of output
+    #' @param settings specific settings for task
+    #' @param active logical indicating if `task` is performed in worklfow.
+    #' Default value is `TRUE`
+    #' @param message title of the `task`.
+    #' Default value indicates `task` name.
+    #' @return A new `Task` object
+    plotGoFSettings = function(input = NULL,

```

Comment:
This corresponds to "Time profiles and residuals" task, correct? https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/issues/18
Then it might be better to call it `TimeProfilesAndResiduals` instead of `GOF`.